framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -7550,12 +7559,21 @@ void VulkanReplayConsumerBase::GetShadowSemaphores(
         for (uint32_t i = 0; i < count; ++i)
         {
             SemaphoreInfo* semaphore_info = object_info_table_.GetSemaphoreInfo(semaphore_infos[i].semaphore);
-            if ((semaphore_info != nullptr) && (semaphore_info->shadow_signaled == true))
+            if (semaphore_info != nullptr)
             {
-                // If found, unsignal the semaphore to represent it being used.
-                shadow_semaphores->push_back(semaphore_info);
-                semaphore_info->shadow_signaled = false;
-                shadow_semaphores_.erase(semaphore_info->handle);
+                if (semaphore_info->shadow_signaled == true)
```

Comment:
Is this getting complex enough that it (and the duplicate code above) should be in a helper function?

Reply:
Done